### PR TITLE
Add dashboard providing obsolete action information

### PR DESCRIPTION
**Pull request by XxshiftxX**
- Add option on `action` command to provide obsolete action data as json format to dashboard.
   - `.Lib9c.Tools/SubCommand/Action.cs`
- Add gh-action workflow generating json data with `action command`
   - `.github/*`
- Add dashboard providing obsolete action information
   - `Dashboard/*`

### CODE REVIEW

File: .github/exports/obsolete-actions.json
Comment (moreal) [inline .github/exports/obsolete-actions.json:1]: Is this file needed? 🤔

File: Dashboard/README.md
Comment (moreal) [inline Dashboard/README.md:1]: Could you rewrite `README.md`? About *How to build?*, *What environment variables required?*, etc...

File: Dashboard/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Vite + React + TS</title>
```
Comment (moreal) [inline Dashboard/index.html:7]:
```suggestion
    <title>Obsolete Dashboard</title>
```

File: Dashboard/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "predeploy": "yarn build",
+    "deploy": "gh-pages -d build",
+    "build": "tsc && vite build",
+    "lint": "eslint . --ext ts,tsx --report-unused-disable-directives --max-warnings 0",
+    "preview": "vite preview"
+  },
+  "dependencies": {
+    "@tanstack/react-query": "^4.32.0",
+    "axios": "^1.4.0",
+    "gh-pages": "^5.0.0",
```
Comment (moreal) [inline Dashboard/package.json:17]: Isn't it `devDependencies`? 🤔

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -104,6 +104,12 @@ jobs:
       run: |
         mkdir publish
         dotnet run --property WarningLevel=0 --project .Lib9c.Tools/Lib9c.Tools.csproj -- action list > publish/all_action_type_ids.txt
+        dotnet run --property WarningLevel=0 --project .Lib9c.Tools/Lib9c.Tools.csproj -- action list --obsolete-only --json-path=publish/obsolete_action_types.json
+    - name: Build Dashboard
+      run: |
+        cd Dashboard
+        yarn && yarn build --outDir ../publish
+        cd ..
```
Comment (moreal) [inline .github/workflows/main.yml:112]: The Dashboard can be deployed independently from the Lib9c's release schedule. How about creating a new job to deploy the Dashboard app?

File: .Lib9c.Tools/SubCommand/Action.cs [hunk 4]
```diff
@@ -22,29 +30,53 @@ public class Action
...
+                    type.GetCustomAttribute<ActionTypeAttribute>() is null)
+                {
+                    return false;
+                }
+
+                var isObsoleteTarget = type.GetCustomAttribute<ActionObsoleteAttribute>() is not null;
+                var isObsolete = type.GetCustomAttribute<ActionObsoleteAttribute>() is { } attr &&
+                                 attr.ObsoleteIndex <= blockIndex;
+
+                if (obsoleteOnly) return isObsoleteTarget;
```
Comment (moreal) [inline .Lib9c.Tools/SubCommand/Action.cs:47]: `isObsoleteTarget` just means that the action has an `ActionObsoleteAttribute` attribute. And it doesn't mean that the action is obsoleted. So it cannot match with the `obsoleteOnly` option's description, `show obsoleted actions only`.

File: .github/exports/obsolete-actions.json
Comment (XxshiftxX) [reply .github/exports/obsolete-actions.json:1]: It was committed to provide json file to my local app. I thought it will be ok to exist cause It will be overwrited when this branch merged in main by gh-actions.

Comment (moreal) [reply .github/exports/obsolete-actions.json:1]: Hmm.. could you let me know what part to overwrite this file? It overwrites files on `gh-pages` branch, doesn't it? 🤔

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -108,12 +108,7 @@ jobs:
       run: |
         mkdir publish
         dotnet run --property WarningLevel=0 --project .Lib9c.Tools/Lib9c.Tools.csproj -- action list > publish/all_action_type_ids.txt
-    - name: Publish available action type ids
-      uses: peaceiris/actions-gh-pages@v3
-      with:
-        github_token: ${{ secrets.GITHUB_TOKEN }}
-        publish_dir: ./publish
-        destination_dir: ${{ github.ref_name }}
```
Comment (moreal) [inline .github/workflows/main.yml:116]: Well... I don't think these lines can be removed 🤔 Does it become useless?

**Review by moreal — COMMENTED** [review]
I will leave an additional review

File: .github/workflows/gh-page.yml [hunk 6]
```diff
@@ -0,0 +1,35 @@
+name: main
+
+on:
+  push:
+    branches:
+      - "development"
+      - "add-obsolete-action-command"
```
Comment (moreal) [inline .github/workflows/gh-page.yml:7]: This line can be removed 👀 (It might be for testing.)

File: .github/workflows/gh-page.yml [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Build Dashboard
+      working-directory: ./Dashboard
+      run: yarn build --outDir ../publish
+
+    - name: Deploy
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./publish
+        destination_dir: ${{ github.ref_name }}
```
Comment (moreal) [inline .github/workflows/gh-page.yml:35]: It seems you intended that the *Obsolete Dashboard*'s url becomes `https://planetarium.github.io/lib9c/development`. RIght?

File: .github/workflows/gh-page.yml [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        submodules: true
+
+    - name: Install Dependency
+      working-directory: ./Dashboard
+      run: yarn
+
+    - name: Build .env
+      run: |
+        echo "VITE_OBSOLETE_DATA_PATH=$VITE_OBSOLETE_DATA_PATH" >> ./Dashboard/.env.production
+        echo "VITE_API_PATH=$VITE_API_PATH" >> ./Dashboard/.env.production
```
Comment (moreal) [inline .github/workflows/gh-page.yml:24]: I don't remember it well... To use `secrets`, it should be:
```suggestion
        echo "VITE_OBSOLETE_DATA_PATH=${{ secrets.VITE_OBSOLETE_DATA_PATH }}" >> ./Dashboard/.env.production
        echo "VITE_API_PATH=${{ secrets.VITE_API_PATH }}" >> ./Dashboard/.env.production
```

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-secrets

File: .github/workflows/gh-page.yml (hunk 6 above)
Comment (XxshiftxX) [reply .github/workflows/gh-page.yml:7]: Yes It's just for testing so it will be removed before merge branch

File: .github/workflows/gh-page.yml (hunk 7 above)
Comment (XxshiftxX) [reply .github/workflows/gh-page.yml:35]: It should be `https://planetarium.github.io/lib9c/dashboard`. Thank you for let me know.

**Review by moreal — APPROVED** [review]
🚀🚀🚀 🙏🏻

**Comment by moreal** [comment]
You should rebase this branch and you can merge this pull request 🙏🏻
